frontend/src/modules/Worksheets/components/WorksheetDocAnalyzer.js

```diff
@@ -0,0 +1,249 @@
+                  endAdornment: (
+                    <>
+                      {loadingEnvs ? (
+                        <CircularProgress color="inherit" size={20} />
+                      ) : null}
+                    </>
+                  )
+                }}
+              >
+                {environmentOptions.map((environment) => (
+                  <MenuItem
+                    key={environment.environmentUri}
+                    value={environment}
+                  >
+                    {environment.label}
+                  </MenuItem>
+                ))}
+              </TextField>
+            </Box>
+            <Box sx={{ p: 2, mt: 2 }}>
+              <TextField
+                disabled
+                fullWidth
+                label="Team"
+                name="team"
+                value={worksheet ? worksheet.SamlAdminGroupName : ''}
+                variant="outlined"
+              />
+            </Box>
+            <Box sx={{ p: 2 }}>
+              <TextField
+                fullWidth
+                label="Owned S3 Bucket(s)"
+                name="database"
+                onChange={(event) => {
+                  handleBucketChange(event);
+                }}
+                select
+                value={selectedDatabase}
+                variant="outlined"
+                InputProps={{
+                  endAdornment: (
+                    <>
+                      {loadingDatabases ? (
+                        <CircularProgress color="inherit" size={20} />
+                      ) : null}
+                    </>
+                  )
+                }}
+              >
+                {filteredDBOptions.length > 0 ? (
+                  filteredDBOptions.map((database) => (
+                    <MenuItem key={database.datasetUri} value={database}>
+                      {database.bucketName}
+                    </MenuItem>
+                  ))
+                ) : (
+                  <MenuItem disabled>No owned buckets found</MenuItem>
+                )}
+              </TextField>
+            </Box>
+            <Box sx={{ p: 2 }}>
+              <TextField
+                fullWidth
+                label="S3 Object Key(s)"
+                name="key"
+                onChange={(event) => {
+                  handleKeyChange(event);
+                }}
+                select
+                value={selectedKey}
+                variant="outlined"
+                InputProps={{
+                  endAdornment: (
+                    <>
+                      {loadingKeys ? (
+                        <CircularProgress color="inherit" size={20} />
+                      ) : null}
+                    </>
+                  )
+                }}
+              >
+                {keyOptions.length > 0 ? (
+                  keyOptions.map((key) => (
+                    <MenuItem key={key} value={key}>
+                      {key}
+                    </MenuItem>
+                  ))
```

Comment:
The list of S3 objects can be pretty long. As next steps we can add search functionality here. But next steps

Reply:
yes next step please 😄 - good enhancement though